src/server/middleware/authentication.ts

```diff
@@ -0,0 +1,108 @@
+import passport from 'passport'
+import passportLocal, { VerifyFunctionWithRequest } from 'passport-local'
+import bcrypt from 'bcryptjs'
+import { Express } from 'express'
+import { FieldValidators, User, UserStatus, Validator } from '@openforis/arena-core'
+
+import { ExpressInitializer } from '../expressInitializer'
+import { UserServiceServer } from '../../service'
+import { UserResetPasswordRepository, AuthGroupRepository } from '../../repository'
+
+const comparePassword = bcrypt.compareSync
+
+// TODO: Handle these separately
+const ValidationMessages = {
+  emailInvalid: 'validationErrors.user.emailInvalid',
+  userNotFound: 'validationErrors.user.userNotFound',
+  passwordChangeRequired: 'validationErrors.user.passwordChangeRequired',
+}
+
+const _initializeUser = async (_user: User): Promise<User> => {
+  const user = { ..._user }
+  if (user.password) delete user.password
+
+  // Assoc auth groups
+  let userUpdated = {
+    ...user,
+    authGroups: await AuthGroupRepository.fetchUserGroups({ userUuid: user.uuid }),
+  }
+  if (user.status === UserStatus.INVITED) {
+    const expired = !(await UserResetPasswordRepository.existsResetPasswordValidByUserUuid({ userUuid: user.uuid }))
+    userUpdated = {
+      ...userUpdated,
+      invitation: {
+        expired,
+      },
+    }
+  }
+
+  // @ts-ignore fixed by implementing repository method
+  return userUpdated
+}
+
+const findUserByEmailAndPassword = async (email: string, password: string) => {
+  const user = await UserServiceServer.get({ email })
+
+  if (user?.password && (await comparePassword(password, user.password))) {
+    return _initializeUser(user)
+  }
+
+  return null
+}
+
+const _verifyCallback: VerifyFunctionWithRequest = async (_, email, password, done) => {
+  const sendUser = (user: User) => done(null, user)
+  const sendError = (message: string) => done(null, false, { message })
+
+  const validator = new Validator()
+
+  const { valid: isValidUser } = await validator.validate(
+    { email },
+    { email: [FieldValidators.email('invalid_email')] }
+  )
+
+  if (!isValidUser) {
+    sendError(ValidationMessages.emailInvalid)
+    return
+  }
+
+  const user = await findUserByEmailAndPassword(email, password)
+
+  if (!user) {
+    sendError(ValidationMessages.userNotFound)
+    return
+  }
+
+  if (user.status === UserStatus.ACCEPTED) {
+    sendUser(user)
+  }
+
+  sendError(ValidationMessages.passwordChangeRequired)
+}
+
+const LocalStrategy = passportLocal.Strategy
```

Comment:
why this? cant you just use passportLocal.Strategy? or import Strategy as LocalStrategy ?